chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -0,0 +1,152 @@
+    }
+}
+
+impl RpcTransactionStatusCommonRequest {
+    pub fn parse(value: Option<Value>) -> Result<Self, crate::errors::RpcParseError> {
+        if let Ok((hash, account_id)) =
+            crate::utils::parse_params::<(near_primitives::hash::CryptoHash, String)>(value.clone())
+        {
+            if !near_runtime_utils::is_valid_account_id(&account_id) {
+                return Err(crate::errors::RpcParseError(format!(
+                    "Invalid account id: {}",
+                    account_id
+                )));
+            }
+            let transaction_info = TransactionInfo::TransactionId { hash, account_id };
+            Ok(Self { transaction_info })
+        } else {
+            // TODO: move this code to crate::utils::parse_signed_transaction
+            let (encoded,) = crate::utils::parse_params::<(String,)>(value.clone())?;
+            let bytes = crate::utils::from_base64_or_parse_err(encoded)?;
+            let signed_transaction =
+                near_primitives::transaction::SignedTransaction::try_from_slice(&bytes).map_err(
+                    |err| {
+                        crate::errors::RpcParseError(format!(
+                            "Failed to decode transaction: {}",
+                            err
+                        ))
+                    },
+                )?;
+            let transaction_info = TransactionInfo::Transaction(signed_transaction);
+            Ok(Self { transaction_info })
+        }
+    }
+}
+
+impl From<near_client_primitives::types::TxStatusError> for RpcTransactionError {
+    fn from(error: near_client_primitives::types::TxStatusError) -> Self {
+        match error {
+            near_client_primitives::types::TxStatusError::ChainError(err) => {
+                Self::InternalError { debug_info: format!("{:?}", err) }
+            }
+            near_client_primitives::types::TxStatusError::MissingTransaction(
+                requested_transaction_hash,
+            ) => Self::UnknownTransaction { requested_transaction_hash },
+            near_client_primitives::types::TxStatusError::InvalidTx(context) => {
+                Self::InvalidTransaction { context }
+            }
+            near_client_primitives::types::TxStatusError::InternalError => {
+                Self::InternalError { debug_info: format!("Internal error") }
+            }
+            near_client_primitives::types::TxStatusError::TimeoutError => Self::TimeoutError,
+        }
+    }
+}
+
+impl From<near_primitives::views::FinalExecutionOutcomeViewEnum> for RpcTransactionResponse {
+    fn from(
+        final_execution_outcome: near_primitives::views::FinalExecutionOutcomeViewEnum,
+    ) -> Self {
+        Self { final_execution_outcome }
+    }
+}
+
+impl From<RpcTransactionError> for crate::errors::RpcError {
+    fn from(error: RpcTransactionError) -> Self {
+        let error_data = match error {
+            RpcTransactionError::InvalidTransaction { ref context } => Some(
```

Comment:
```suggestion
        let error_data = match &error {
            RpcTransactionError::InvalidTransaction { context } => Some(
```

Seems simpler to me personally.